OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -101,7 +103,9 @@ void INotifyCreated.Created(Actor self)
 
 		void INotifyDamage.Damaged(Actor self, AttackInfo e)
 		{
-			if (Info.UncloakOn.HasFlag(UncloakType.Damage))
+			var isHeal = e.Damage.Value < 0;
```

Comment:
Sounds like an out-of-scope thing for this PR, because `HarvesterAttackNotifier` uses `INotifyDamage` to trigger EVA warnings and your proposition would prevent it from triggering on pure-status warheads thenafter (from what I tested, targetvalidation still needs a dummy damagewarhead atm).